### PR TITLE
Update User Guide - Multivariable charts and the text configuration

### PR DESCRIPTION
**Pull request by ElenaGallo**
## Description

Add Multivariable charts on [Widgets](https://mapstore--9677.org.readthedocs.build/en/9677/user-guide/widgets/#chart) section,  [Add Advanced options for Pie Chart and Counter#9611](https://github.com/geosolutions-it/MapStore2/issues/9611) with new ui and add and the new text configuration

### CODE REVIEW

File: docs/user-guide/widgets.md [hunk 1]
```diff
@@ -67,27 +93,30 @@ Just below the chart's preview, the following configurations are available:
...
 
-* Change the default **Color** of bars or slices (depending on the chart type) through the *Color Picker*. This color will be applied for all values for which a *Class Color* has not been configured.
+###### Classification style
 
-<video class="ms-docimage" controls><source src="../img/widgets/custom_colors.mp4"/></video>
+For *Bar Charts* and *Pie Charts*, [MapStore](https://mapstore.geosolutionsgroup.com/mapstore/#/) provides the possibility to customize the colors of the charts bars and slices. It is possible to classify the style based on the attributes of the layer. In this case it is possible to select the `Attribute` to use for the classification along with the classification `Method` (*Quantile*, *Equal interval*, *Natural breaks* and *Unique interval*) and select whether to `Sort` the classification by *Attribute X* or *Attribute Y*. Choose a `Color ramp`, the number of classification `Intervals` and the order (with `Reverse order`) of the classification intervals colors. Finally customize the `Outline color` and the `Outline width`.
+
+<img src="../img/widgets/trace_classification_style.jpg" class="ms-docimage"  style="max-width:450px;"/>
```
Comment (tdipisa) [inline docs/user-guide/widgets.md:112]: Please use a chart where the stacked values are better represented in a nicer way. For example the one available [here](https://dev-mapstore.geosolutionsgroup.com/mapstore/#/dashboard/21694)

image

File: docs/user-guide/widgets.md [hunk 2]
```diff
@@ -67,27 +93,30 @@ Just below the chart's preview, the following configurations are available:
...
 
-* Select an *Attribute* in the dropdown list as a **Classification attribute**.
+From the **Color Ramp** option, the user can select the *Custom colors* option and open the **Custom Colors Settings** modal through the <img src="../img/button/edit_button.jpg" class="ms-docbutton"/> button.
+Inside this modal, the user is allowed to:
 
-<video class="ms-docimage" controls><source src="../img/widgets/classification-attribute.mp4"/></video>
+* Change the default **Color** of bars or slices (depending on the chart type) through the *Color Picker*. This color will be applied for all values for which a *Class Color* has not been configured.
 
-Once the attribute is chosen, new options appear in the *Custom Color Settings* panel that allow the user to:
+<video class="ms-docimage" controls><source src="../img/widgets/custom_colors.mp4"/></video>
```
Comment (tdipisa) [inline docs/user-guide/widgets.md:119]: Same of [this](https://github.com/geosolutions-it/MapStore2/pull/9677/files#diff-fa9572d42754707058739c40bac414046b2e986d1dbf35a065d141e26fe71bf7R112) here.

File: docs/user-guide/widgets.md [hunk 3]
```diff
@@ -67,27 +93,30 @@ Just below the chart's preview, the following configurations are available:
...
 
 * Enter a **Default Class Label** to be used in the legend for all values that will not be specifically classified in the following list.
```
Comment (tdipisa) [inline docs/user-guide/widgets.md:121]: Same of [this](https://github.com/geosolutions-it/MapStore2/pull/9677/files#diff-fa9572d42754707058739c40bac414046b2e986d1dbf35a065d141e26fe71bf7R112) here.

File: docs/user-guide/widgets.md [hunk 4]
```diff
@@ -98,101 +127,147 @@ Once the attribute is chosen, new options appear in the *Custom Color Settings*
...
 
-* **Stacked**. An example can be the following:
+<img src="../img/widgets/custom_trace_value_tooltip.jpg" class="ms-docimage"  style="max-width:450px;"/>
 
-<img src="../img/widgets/stacked-chart.jpg" class="ms-docimage" style="max-width:450px;"/>
+##### Trace legend options
 
-!!! Note
-    By default, the bar chart type is **Stacked**
+For the *Pie Charts*, the *Trace legend options* is available and is displayed as follows:
```
Comment (tdipisa) [inline docs/user-guide/widgets.md:185]:
```suggestion
For the *Pie Charts*, the *Trace legend options* is available and it is displayed as follows:
```

File: docs/user-guide/widgets.md [hunk 5]
```diff
@@ -98,101 +127,147 @@ Once the attribute is chosen, new options appear in the *Custom Color Settings*
...
+The user can:
+
+* Show/Hide the percentages in legend with the **Include percentages in legend** control. An example with the percentages in legend can be the following:
+
+<img src="../img/widgets/percentages_in_legend.jpg" class="ms-docimage"  style="max-width:450px;"/>
+
+#### Axes
 
-#### Advanced Options
+[MapStore](https://mapstore.geosolutionsgroup.com/mapstore/#/) allows to customize the **Y axis** and the **X axis** for the *Bar Charts* and the *Line Charts* through the *Axes*.
```
Comment (tdipisa) [inline docs/user-guide/widgets.md:197]:
```suggestion
[MapStore](https://mapstore.geosolutionsgroup.com/mapstore/#/) allows to customize the **Y axis** and the **X axis** for the *Bar Charts* and the *Line Charts* through the *Axes* tab.
```

File: docs/user-guide/widgets.md [hunk 6]
```diff
@@ -98,101 +127,147 @@ Once the attribute is chosen, new options appear in the *Custom Color Settings*
...
+<img src="../img/widgets/axes_tab.jpg" class="ms-docimage"/>
 
-<img src="../img/widgets/advanced_options.jpg" class="ms-docimage"/>
+Through this section, for each axis, the user is allowed to:
 
-Through this section, the user is allowed to:
+* *Edit* the current **Axis Title** through the <img src="../img/button/edit_button.jpg" class="ms-docbutton"/> button
 
-* Show/Hide the chart's grid in backgroung with the **Hide Grid** control
+* If there is more than one [Traces](#traces), *Add new axis* through the <img src="../img/button/+++.jpg" class="ms-docbutton"/> button
```
Comment (tdipisa) [inline docs/user-guide/widgets.md:205]:
```suggestion
* If there is more than one [Trace](#traces), *Add new axis* through the <img src="../img/button/+++.jpg" class="ms-docbutton"/> button
```

File: docs/user-guide/widgets.md [hunk 7]
```diff
@@ -98,101 +127,147 @@ Once the attribute is chosen, new options appear in the *Custom Color Settings*
...
 
-* Customize **Y axis** tick values by choosing the *Type* (between `Auto`, `Linear`, `Category`, `Log` or `Date`): the axis type is auto-detected by looking at data (*Auto* option is automatically managed and selected by the tool and it is usually good as default setting). The user can also choose to completely hide labels through the *Hide labels* control or customize them by adding a *Prefix* (e.g. `~`), a custom *Format* (e.g. `0%: rounded percentage, '12%'` or more) or a *Suffix* (e.g. `Km`). It is also possible to configure a *Formula* to transform tick values as needed (e.g. `value + 2` or `value / 100` or more)
+* If there is more than one [Traces](#traces), *Delete* the current axis through the <img src="../img/button/delete_button.jpg" class="ms-docbutton"/> button
```
Comment (tdipisa) [inline docs/user-guide/widgets.md:207]:
```suggestion
* If there is more than one [Trace](#traces), *Delete* the current axis through the <img src="../img/button/delete_button.jpg" class="ms-docbutton"/> button
```

File: docs/user-guide/widgets.md [hunk 8]
```diff
@@ -98,101 +127,147 @@ Once the attribute is chosen, new options appear in the *Custom Color Settings*
...
 
-<img src="../img/widgets/yaxis_options.jpg" class="ms-docimage"/>
+* Choose the **Type** (between `Auto`, `Linear`, `Category`, `Log` or `Date`): the axis type is auto-detected by looking at the data (*Auto* option is automatically managed and selected by the tool and it is usually good as the default setting).
```
Comment (tdipisa) [inline docs/user-guide/widgets.md:209]:
```suggestion
* Choose the **Type** (between `Auto`, `Linear`, `Category`, `Log` or `Date`): the axis type is auto-detected by looking at the data (*Auto* option is automatically managed and selected by the tool and it is usually good as default setting).
```

File: docs/user-guide/widgets.md [hunk 9]
```diff
@@ -98,101 +127,147 @@ Once the attribute is chosen, new options appear in the *Custom Color Settings*
...
-    The tick labels available for the X axis by enabling the option **Never skip label** cannot be more than 200 in order to provide a clear chart and for performance reasons.
+    The *Never skip label* option is available only for the *X axis* and the tick labels available by enabling the option cannot be more than 200 in order to provide a clear chart and for performance reasons.
+
+* Enable **Label rotation** to better adapt axis tick labels on the charts depending on the needs.
+
+* Choose to completely hide labels through the **Hide labels** control.
+
+#### Layout
+
+Through the *Layout* tab, the user can also customize the **Layout** of the chart and the default **Font** for all the texts present in the chart.
```
Comment (tdipisa) [inline docs/user-guide/widgets.md:237]:
```suggestion
Through the *Layout* tab, the user can also customize the **Layout** of the chart and the default **Font** for all the labels present in the chart.
```

**Review by tdipisa — CHANGES_REQUESTED** [review]

File: docs/user-guide/widgets.md (hunk 1 above)
Comment (tdipisa) [reply docs/user-guide/widgets.md:112]: @ElenaGallo what about this one?

File: docs/user-guide/widgets.md (hunk 3 above)
Comment (tdipisa) [reply docs/user-guide/widgets.md:121]: @ElenaGallo what about this one?

File: docs/user-guide/widgets.md (hunk 1 above)
Comment (ElenaGallo) [reply docs/user-guide/widgets.md:112]: Added a new image with commit: [review_9669](https://github.com/geosolutions-it/MapStore2/pull/9677/commits/a4148f53c053eea7453c8fe5387910d2ccc2d122)

File: docs/user-guide/widgets.md (hunk 2 above)
Comment (ElenaGallo) [reply docs/user-guide/widgets.md:119]: Added a new video with commit: [review_9669](https://github.com/geosolutions-it/MapStore2/pull/9677/commits/a4148f53c053eea7453c8fe5387910d2ccc2d122)

File: docs/user-guide/widgets.md (hunk 3 above)
Comment (ElenaGallo) [reply docs/user-guide/widgets.md:121]: Added a new image with commit: [review_9669](https://github.com/geosolutions-it/MapStore2/pull/9677/commits/a4148f53c053eea7453c8fe5387910d2ccc2d122)

File: docs/user-guide/widgets.md (hunk 1 above)
Comment (tdipisa) [reply docs/user-guide/widgets.md:112]: @ElenaGallo correct if I'm wrong but this change doesn't appear in the doc build:
https://mapstore--9677.org.readthedocs.build/en/9677/user-guide/widgets/

did I miss something?

File: docs/user-guide/widgets.md (hunk 2 above)
Comment (tdipisa) [reply docs/user-guide/widgets.md:119]: @ElenaGallo correct if I'm wrong but this change doesn't appear in the doc build:
https://mapstore--9677.org.readthedocs.build/en/9677/user-guide/widgets/

did I miss something?

File: docs/user-guide/widgets.md (hunk 3 above)
Comment (tdipisa) [reply docs/user-guide/widgets.md:121]: @ElenaGallo correct if I'm wrong but this change doesn't appear in the doc build:
https://mapstore--9677.org.readthedocs.build/en/9677/user-guide/widgets/

did I miss something?

**Review by tdipisa — CHANGES_REQUESTED** [review]
Beside comments above it seems to me that also the documentation for this one should be included with this PR (you have connected the issue) 
https://github.com/geosolutions-it/MapStore2/issues/9816

but that doc is still missing. Right?

**Comment by ElenaGallo** [comment]
@tdipisa about this https://github.com/geosolutions-it/MapStore2/pull/9677#pullrequestreview-1942594741:

I had the doc for #9816 [here](https://github.com/geosolutions-it/MapStore2/pull/9677/commits/0664c327de4f10aac8c7d31fd89c080bdbfa878e) but I improved it in this latest commit [here](https://github.com/geosolutions-it/MapStore2/pull/9677/commits/15253580f81168e4434c90398eaa18906faf87ff) . Can it be clear this way?

File: docs/user-guide/widgets.md [hunk 10]
```diff
@@ -37,6 +37,14 @@ Selecting a *Layer* or *Layers*, the following *Chart* options is presented to u
 
 <img src="../img/widgets/chart-options.jpg" class="ms-docimage"  style="max-width:450px;"/>
 
+From the top toolbar of this panel the user is allowed to:
```
Comment (tdipisa) [inline docs/user-guide/widgets.md:40]: @ElenaGallo if you want to specify this at table widget level and chart level why you shouldn't do that also for counter widgets?

File: docs/user-guide/widgets.md [hunk 11]
```diff
@@ -345,7 +347,17 @@ Adding a table widget to the map, a panel like the following opens:
 
 <img src="../img/widgets/table-panel.jpg" class="ms-docimage"  style="max-width:450px;"/>
 
-The toolbar on the top of this panel is similar to the one present in [Chart section](#chart). Here the user is allowed to:
+From the top toolbar of this panel the user is allowed to:
```
Comment (tdipisa) [inline docs/user-guide/widgets.md:350]: @ElenaGallo if you want to specify this at table widget level and chart level why you shouldn't do that also for counter widgets?

File: docs/user-guide/filtering-layers.md [hunk 12]
```diff
@@ -139,6 +139,9 @@ This tool is used to define advanced filters in [MapStore](https://mapstore.geos
 
 <img src="../img/filtering-layers/query-panel.jpg" class="ms-docimage" style="max-width:500px;"/>
 
+!!! note
+    For configuring layer filtering in [Widgets](widgets.md) the **Layer Filter** option is only available for [Maps](exploring-maps.md) and not for [Dashboards](exploring-dashboards.md)
```
Comment (tdipisa) [inline docs/user-guide/filtering-layers.md:143]:
```suggestion
    The **Layer Filter** option is only available for [Maps](exploring-maps.md) and not for [Dashboards](exploring-dashboards.md)
```

**Review by tdipisa — CHANGES_REQUESTED** [review]

**Comment by tdipisa** [comment]
> @tdipisa about this [#9677 (review)](https://github.com/geosolutions-it/MapStore2/pull/9677#pullrequestreview-1942594741):
> 
> I had the doc for #9816 [here](https://github.com/geosolutions-it/MapStore2/pull/9677/commits/0664c327de4f10aac8c7d31fd89c080bdbfa878e) but I improved it in this latest commit [here](https://github.com/geosolutions-it/MapStore2/pull/9677/commits/15253580f81168e4434c90398eaa18906faf87ff) . Can it be clear this way?

I'm sorry @ElenaGallo a couple of things:


- See my review here https://github.com/geosolutions-it/MapStore2/pull/9677#pullrequestreview-1943265920

- The aim of https://github.com/geosolutions-it/MapStore2/issues/9098 is to provide spatial filter capabilities also for dashboard widgets. This doesn't seems to be documented.
image

File: docs/user-guide/filtering-layers.md [hunk 13]
```diff
@@ -182,6 +182,10 @@ Once this filter is set, it is always possible to edit the coordinates and the d
 
 <img src="../img/filtering-layers/edit_geom.jpg" class="ms-docimage" style="max-width:500px;"/>
 
+!!! note
+     For [Dashboard](exploring-dashboards.md) [widgets](widgets.md) (charts, table and counter) the **Region of Interest** option is called **Area of interest**. With this type of filter the user can define a spatial filter without connecting the widget to the map widget. An example can be the following:
```
Comment (tdipisa) [inline docs/user-guide/filtering-layers.md:186]:
```suggestion
     Also for [Dashboard](exploring-dashboards.md) [widgets](widgets.md) (charts, table and counter) it is possible to define a spatial filter without necessarily connect the widget to the map widget by using the usual **Area of interest** filtering section. The example below sows how:
```

Comment (tdipisa) [reply docs/user-guide/filtering-layers.md:186]: @ElenaGallo where **Region of Interest** appear in the UI? I see "Area of interest" everywhere. It seems the documentation need to be fixed.

File: docs/user-guide/filtering-layers.md [hunk 14]
```diff
@@ -204,3 +208,6 @@ In order to better understand this type of filter, let's make an example. We sup
 In particular, if our goal is to take a look at the Italian Regions that contain the Unesco sites with *serial code=1*, the operations to perform can be the following:
 
 <video class="ms-docimage" controls><source src="../img/filtering-layers/layer_filter.mp4"/></video>
+
+!!! note
+    In the [widgets](widgets.md) tool, the **Layer Filter** option is only available for [Maps](exploring-maps.md) and not for [Dashboards](exploring-dashboards.md).
```
Comment (tdipisa) [inline docs/user-guide/filtering-layers.md:213]:
```suggestion
    The **Layer Filter** option is only available for [widgets](widgets.md) defined in [Map viewer](exploring-maps.md) and not for [Dashboards](exploring-dashboards.md) widgets.
```

**Review by tdipisa — CHANGES_REQUESTED** [review]

**Review by tdipisa — APPROVED** [review]

**Comment by tdipisa** [comment]
@ElenaGallo please backport to 2024.01.xx. Thank you.

**Pull request merged**